pkg/trace/writer/trace.go

```diff
@@ -215,25 +228,31 @@ func (w *TraceWriter) resetBuffer() {
 
 const headerLanguages = "X-Datadog-Reported-Languages"
 
+// w must be locked for a flush.
```

Comment:
do we need to add locking to `Stop` now that waiting for the waitgroup won't prevent concurrent access to the `writeChunks`? 
ie. can there be a race between one goroutine calling `WriteChunks` (modifying w.tracerPayloads) while a separate go routine is calling stop and flushing? (maybe this isn't a "real" issue and we can instead do the synchronization at the level above this such that Stop is only called once all the processing goroutines are done?)